### PR TITLE
adds credits to the resources of description

### PR DESCRIPTION
**Pull request by yati1998**

### CODE REVIEW

**Comment by yati1998** [comment]
@alodahl I have added the required credits. Please do check

**Comment by yati1998** [comment]
@alodahl Please do review my PR so that I can start working on other issues. Thank you!

**Comment by yati1998** [comment]
@alodahl I have made the required changes. Please do review them

File: roles-in-tech/developer-advocate.md [hunk 1]
```diff
@@ -44,13 +41,16 @@ Steps towards a better Developer Evangelist:
...
 ### Lessons
+
 - Read through [these](https://docs.google.com/presentation/d/1JF2t4VCznUzD2tyW9kuc0Gx8-q0rnpzjDpXxpMFh8DA/edit#slide=id.p) slides.
 
 Inorder to be a developer advocate one must have the following skills:
 - Great development skills — both in creating and explaining software and products
 - Excellent communication skills — this job is about reaching out, listening and distilling information
 - Excellent networking skills — you’re meant to collect contacts and keep them happy
 
+#### [What does a developer evangelist advocate do? By: Christian Heilmann](https://medium.com/@codepo8/what-does-a-developer-evangelist-advocate-do-3ad936dff3e4)
```
[the remark this answers is not in the document]
Comment (alodahl) [reply roles-in-tech/developer-advocate.md]: can you change this to 
do? (By Christian Heilmann)

File: roles-in-tech/developer-advocate.md [hunk 2]
```diff
@@ -65,15 +65,15 @@ Before you jump on the opportunity, there are some very important points to reme
...
 3) Being a consultative source of information for managers within the company to help them understand the importance of the projects to which you are assigned and the role that open-source plays within it.
 4) A participant in product-focused development planning and sprint meetings as an advocate for developer customers to ensure that issues in the product which result in bugs being opened in open-source software that uses your product or are brought to your attention within the developer community are addressed appropriately.
 5) Integration into a product team to assist them with understanding the developer customer use case for the product and driving that direction in API development, documentation, feature development, and test processes.
 
-#### Tasks that you need to do
+#### Tasks that you need to do: By: [Christian Heilmann](https://medium.com/@codepo8/what-does-a-developer-evangelist-advocate-do-3ad936dff3e4)
```
Comment (alodahl) [inline roles-in-tech/developer-advocate.md]: need to do (from "Name of Article" byChristian Heilmann)

File: roles-in-tech/developer-advocate.md [hunk 3]
```diff
@@ -65,15 +65,15 @@ Before you jump on the opportunity, there are some very important points to reme
 It’s a versatile, morphing and evolving role. 
 
  
- #### Roles and Responsibilities
+ #### Roles and Responsibilities: By-[Tyler Duzan](https://www.quora.com/What-exactly-is-the-job-of-a-developer-advocate)
```
Comment (alodahl) [inline roles-in-tech/developer-advocate.md]: (by Tyler Duzan)
no : or - necessary

File: roles-in-tech/machine-learning.md [hunk 4]
```diff
@@ -11,14 +11,14 @@
...
 
 ### Motivation
 
 - Machine learning(ML) is an application of artificial intelligence that provides systems the ability to automatically learn and improve from experience without being explicitly programmed.
 - Machine learning is given so much importance because it helps in predicting behavior and recognizing patterns that humans with their limited capacity cannot predict. It is important because every field wants automatic, adaptive, optimize and want to take the decision itself without any human interface.
 
-**Which companies use Machine Learning?**   
+**Which companies use Machine Learning? From- [em360tech.com](https://www.em360tech.com/tech-news/top-ten/top-10-companies-using-machine-learning/)**
```
Comment (alodahl) [inline roles-in-tech/machine-learning.md]: please change this to 
Learning? (from em360tech.com)

**Comment by yati1998** [comment]
@alodahl I have made the required changes. I hope it works this time.

File: api/REST-API.md [hunk 5]
```diff
@@ -33,46 +33,49 @@ Today there are no projects or applications that don't have a REST API for the c
 - [RESTful API quick guide](https://www.tutorialspoint.com/restful/restful_quick_guide.htm)
    - Go through the Introduction, Resources, Methods, Statelessness and question-answer part of the tutorial.
  
-### Lesson
+### Lesson[(from bbaopen4u)](https://bbvaopen4u.com/en/actualidad/rest-api-what-it-and-what-are-its-advantages-project-development)
```
Comment (alodahl) [inline api/REST-API.md]: By this, you're saying that EVERYTHING in the lesson is from that website.  is this true?  If not, change it to citing the source after the section like in the purdue docs.

File: api/REST-API.md [hunk 6]
```diff
@@ -33,46 +33,49 @@ Today there are no projects or applications that don't have a REST API for the c
...
 
-To give a simple definition, REST is any interface between systems using HTTP to obtain data and generate operations on those data in all possible formats, such as XML and JSON. This is an increasingly popular alternative to other standard data exchange protocols such as SOAP (Simple Object Access Protocol), which have a high capacity but are also very complex. Sometimes it's preferable to use a simpler data-processing solution such as REST. 
+    To  give  a  simple  definition,  REST  is  any  interface  between  systems  using  HTTP  to  obtain  data  and  generate  operations  on  those data in  all  possible  formats,  such  as  XML  and  JSON.  This  is  an  increasingly  popular  alternative  to  other  standard  data  exchange protocols such as SOAP (Simple Object Access Protocol), which have a high capacity but are also very complex. Sometimes it's preferable to use a simpler  data-processing  solution  such  as  REST.
```
Comment (alodahl) [inline api/REST-API.md]: All of your changes are on code notation in this file.  please change it to regular text.

File: github-collaboration/github-collaboration.md [hunk 7]
```diff
@@ -90,7 +90,7 @@ In the industry, apps are created by teams of developers rather than individual
     git push origin master
     ```
     - Now your fork is up to date with the original Techtonica/curriculum repo. For the moment, at least.
- 1. Adding a new remote repo
+ 1. Adding a new remote repo[(from atlassain.com)](https://www.atlassian.com/git/tutorials/syncing)
```
Comment (alodahl) [inline github-collaboration/github-collaboration.md]: you misspelled atlassian

File: roles-in-tech/developer-advocate.md [hunk 8]
```diff
@@ -44,13 +41,16 @@ Steps towards a better Developer Evangelist:
```
Comment (alodahl) [inline roles-in-tech/developer-advocate.md]: Can you remove the quotes here and add the name of the slideshow?

File: api/REST-API.md (hunk 6 above)
Comment (yati1998) [reply api/REST-API.md]: I did so as it was mention in the link that you shared. Do write in this way. Still, if you say I will change it

File: api/REST-API.md (hunk 5 above)
Comment (yati1998) [reply api/REST-API.md]: @alodahl Most of the things are from this website. and for that reason, I have added that in the header. Will it be a problem?

File: roles-in-tech/developer-advocate.md (hunk 8 above)
Comment (yati1998) [reply roles-in-tech/developer-advocate.md]: @alodahl  do you mean the name of the book? there is no slideshow. Shall I mention the book name and add a link to it?

File: api/REST-API.md (hunk 6 above)
Comment (alodahl) [reply api/REST-API.md]: OH, no I just meant please add the citation reference at the end the same way.  sorry.

Comment (yati1998) [reply api/REST-API.md]: Ohk I will change that again.

**Comment by praveenscience** [comment]
@yati1998 It would be easier if you use Typora or something else to write the content, as it shows you a WYSIWYG MarkDown editor that supports GitHub Flavoured Markdown... 😇

File: roles-in-tech/developer-advocate.md (hunk 8 above)
Comment (alodahl) [reply roles-in-tech/developer-advocate.md]: ["Link to the SlideShare:"](https://docs

File: roles-in-tech/developer-advocate.md [hunk 9]
```diff
@@ -37,7 +37,7 @@ Steps towards a better Developer Evangelist as mentioned in a [developer evangel
 ### Materials
 
 
-["Link to the SlideShare:"](https://docs.google.com/presentation/d/1JF2t4VCznUzD2tyW9kuc0Gx8-q0rnpzjDpXxpMFh8DA/edit#slide=id.p)
+[Developer Evangelism](https://docs.google.com/presentation/d/1JF2t4VCznUzD2tyW9kuc0Gx8-q0rnpzjDpXxpMFh8DA/edit#slide=id.p)
```
Comment (alodahl) [inline roles-in-tech/developer-advocate.md]: Techtonica Slideshow: Roles in Tech - Developer Evangelist

File: api/REST-API.md [hunk 10]
```diff
@@ -14,9 +14,10 @@ Here are links to lessons that should be completed before this lesson:
 
 
 ### Motivation
+[From bbvaopen4u](https://bbvaopen4u.com/en/actualidad/rest-api-what-it-and-what-are-its-advantages-project-development)
```
Comment (alodahl) [inline api/REST-API.md]: 🏅

File: api/REST-API.md [hunk 11]
```diff
@@ -35,44 +36,48 @@ Today there are no projects or applications that don't have a REST API for the c
...
+- Uniform interface – By  applying  the  software  engineering  principle  of  generality  to  the  component  interface,  the  overall  system architecture  is  simplified  and  the  visibility  of  interactions  is  improved.  In  order  to  obtain  a  uniform  interface,  multiple architectural  constraints  are  needed  to  guide  the  behavior  of  components.  REST  is  defined  by  four  interface  constraints: identification  of  resources;  manipulation  of  resources  through  representations;  self-descriptive  messages;  and,  hypermedia  as  the engine  of  application  state.
+- Layered  system – The  layered  system  style  allows  an  architecture  to  be  composed  of  hierarchical  layers  by  constraining  component behavior  such  that  each  component  cannot  “see”  beyond  the  immediate  layer  with  which  they  are  interacting.
+- Code  on  demand(optional) – REST  allows  client  functionality  to  be  extended  by  downloading  and  executing  code  in  the  form  of applets  or  scripts.  This  simplifies  clients  by  reducing  the  number  of  features  required  to  be  pre-implemented.
```
Comment (alodahl) [inline api/REST-API.md]: Why is this optional?

**Comment by yati1998** [comment]
@alodahl I have made the changes . Please do review it.

**Pull request merged**